### DISCUSSION TITLE
Is full page screenshot supported?

### DISCUSSION THREAD

**Discussion by ifx-querido** [firecrawl/firecrawl#633] [Q&A] [answered]
According to the [welcome page](https://docs.firecrawl.dev/v1-welcome) of the V1 documentation , the fullpage screenshot is an existing option for the scrape function. Additionaly, @nickscamara just posted a new announcement about the V1, which includes the fullpage screenshot as an [added feature](https://github.com/mendableai/firecrawl/pull/504).

However, the [Advanced Scraping Guide](https://docs.firecrawl.dev/advanced-scraping-guide) only lists ["markdown", "links", "html", "rawHtml", "screenshot"] as output formats. And when I try to run:

```
page_content = app.scrape_url(website, {
  "formats": ['markdown', 'screenshot@fullPage'],
})
print(page_content['screenshot@fullPage'])
```

I get this error:
`KeyError: 'screenshot@fullPage'`

Could someone please clarify this inconsistency?

**Comment by ifx-querido** [comment, marked as answer]
I figured this out on my own.
- In order to get a screenshot of the **top part** of a website, the format option should be specified as "**screenshot**".
- In order to get a **full page** screenshot, the format option should be specified as "**screenshot@fullPage**".
- Either way, to get the screenshot link, the print request **must** be always:
`print(page_content['screenshot'])`
And the link will direct you to the screenshot, which is of the entire page or just the top section, depending on what you put as a format option.

**Reply by robertnowell** [reply]
what is the difference between format and actions with respect to screenshot?

in node:
```
    const crawlResponse = await firecrawl_app.scrapeUrl(url, {
      formats: ["markdown", "html", "screenshot@fullPage"],
      actions: [{ type: "screenshot", fullPage: true }],
    });

```

what is the difference between formats and actions for getting a screenshot?

<img width="931" alt="Screenshot 2025-02-17 at 12 17 13 PM" src="https://github.com/user-attachments/assets/24dec18e-4578-420f-8816-d6c63dafb128" />

**Reply by mogery** [reply]
> what is the difference between format and actions with respect to screenshot?

`screenshot@fullPage` format always runs after all actions. With the screenshot action, you can take multiple screenshots in between e.g. clicking buttons. The results will be in `actions.screenshots` in the response.

**Comment by robertnowell** [comment]
thank you! very helpful. Is this explained in depth in the documentation
somewhere? I didn’t understand from reading through the main docs

Find me on LinkedIn <http://www.linkedin.com/in/robert-nowell>, Github
<https://github.com/robertnowell>

**Reply by mogery** [reply]
Just extended the API reference to show this info on the actions Screenshot tab as well.
